src/app/view/components/login/dot-login-component/dot-login.component.ts

```diff
@@ -0,0 +1,164 @@
+        this.disableFormFields(value);
+    }
+    @Output() login = new EventEmitter<DotLoginCredentials>();
+
+    @ViewChild('languageDropdown') languageDropdown: Dropdown;
+
+    loginForm: FormGroup;
+    languages: SelectItem[] = [];
+
+    loginInfo$: Observable<DotLoginInformation>;
+
+    private languageChange$: Subject<DotLoginInformation> = new Subject<DotLoginInformation>();
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private loginService: LoginService,
+        private fb: FormBuilder,
+        private route: ActivatedRoute,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    ngOnInit() {
+        this.loginService.setAuth({
+            loginAsUser: null,
+            user: null,
+            isLoginAs: false
+        });
+
+        this.loginForm = this.fb.group({
+            login: ['', [Validators.required]],
+            language: [''],
+            password: ['', [Validators.required]],
+            rememberMe: false
+        });
+
+        const onLanguageChange = this.languageChange$
+            .asObservable()
+            .pipe(
+                takeUntil(this.destroy$),
+                switchMap((event: any) =>
+                    this.loginService
+                        .getLoginFormInfo(event, LOGIN_LABELS)
+                        .pipe(map(loginInfo => ({ loginFormInfo: loginInfo })))
+                )
+            );
+
+        this.loginInfo$ = merge(this.route.data, onLanguageChange).pipe(
+            takeUntil(this.destroy$),
+            pluck('loginFormInfo'),
+            map((loginInfo: DotLoginInformation) => this.setUserNameLabel(loginInfo)),
+            tap((loginInfo: DotLoginInformation) => {
+                this.setInitialFormValues(loginInfo);
+            })
+        );
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+    /**
+     *  Executes the logIn service
+     */
+    logInUser(): void {
+        if (this.loginForm.valid) {
+            this.login.emit(this.loginForm.value);
+        }
+    }
+
+    /**
+     * Display the forgot password card
+     */
+    showForgotPassword(): void {
+        this.dotRouterService.goToForgotPassword();
+    }
+
+    private setInitialFormValues(loginInfo: DotLoginInformation): void {
+        this.loginForm
+            .get('language')
+            .setValue(this.setLanguageFormat(loginInfo.entity.currentLanguage));
+        this.setLanguageItems(loginInfo);
+        this.checkMessages(loginInfo);
+    }
+
+    private checkMessages(loginInfo: DotLoginInformation): void {
+        if (this.passwordChanged) {
+            this.message = loginInfo.i18nMessagesMap['reset-password-success'];
+        }
+        if (this.resetEmailSent) {
+            this.message = loginInfo.i18nMessagesMap['a-new-password-has-been-sent-to-x'].replace(
+                '{0}',
+                this.resetEmail
+            );
+        }
```

Comment:
message is a input so I think we should not change this here, this case should be handle be the container and just set the right message in the component